apps/sms/js/navigation.js

```diff
@@ -268,16 +285,16 @@ var Navigation = {
       }.bind(this),
       function rejected(e) {
         catchError(e);
-        this.transitioning = false;
+        this.transitionPromise = null;
         return Promise.reject(new Error('Error while transitioning'));
       }.bind(this)
     );
 
-    promise.
+    this.transitionPromise.
       then(() => this.emit('navigated', { panel: panel, args: args })).
       then(nextQueuedPanel, nextQueuedPanel);
 
-    return promise;
+    return this.transitionPromise;
```

Comment:
could lead to less change if you kept the local variable `promise` and finish with `return this.transitionPromise = promise`.

This would work the same and could make the code slightly more readable.